src/services/questionnaireAnalyzer/questionnaireAnalyzer.js

```diff
@@ -398,63 +398,44 @@ const checkCompletionStateOfMultipleItems = (items, props) => {
 
 	let props = store.getState().CheckIn
-	
-	//if item is supposed to be hidden
-	if(item.extension && item.extension[0].valueBoolean && item.extension[0].valueBoolean === true){
-		returnValue = false
-	}
-	// if the item has a set of conditions
-	else if (item && item.enableWhen) {
 
+	//if item is supposed to be hidden
+	let hiddenExtension = item.extension?.find(it => it.url === "http://hl7.org/fhir/StructureDefinition/questionnaire-hidden");
+	if (hiddenExtension && hiddenExtension.valueBoolean === true) {
+		returnValue = false;
+	} else if (item && item.enableWhen) { // if the item has a set of conditions
 		// checks if the items mentioned in the conditions are even answered...
-		if (!checkIfAnswersToConditionsAreAvailable (item )) {
+		if (!checkIfAnswersToConditionsAreAvailable(item)) {
+			console.log("Answers to condition are not available!!!", item.linkId)
 			// ...if not, the returnValue is set to FALSE - game over
 			returnValue = false
-		}
-		else {
-			// if no enableBehavior is set (or it is set to "all")
-			if(!item.enableBehavior || item.enableWhen.length === 0 || (item.enableBehavior && item.enableBehavior === 'all')) {
-				
-				// sets the default to TRUE, as one non-matching condition is enough to turn the result FALSE
-				returnValue = true
-				
-				// iterates over all conditions
-				item.enableWhen.forEach(condition => {
-					// triggers if NO MATCH is found 
-					if (!(
-							// if the condition provides an array of answers and the current answer is among then
-							(Array.isArray(props.questionnaireItemMap[condition.question].answer) && props.questionnaireItemMap[condition.question].answer.includes( condition[getEnableWhenAnswerType(condition)])) 
-							||
-							// OR: there is only one answer and it matches
-							(getCorrectlyFormattedAnswer(props.questionnaireItemMap[condition.question]) === condition[getEnableWhenAnswerType(condition)])
-						)
-					){
-						// in case a not-matching condition is found
-						returnValue = false
-					}
-				})
-			}
-			// if enableBehavior is set to 'any'
-			else {
-				// iterates over all conditions
-				item.enableWhen.forEach(condition => {
-					if (
-						// if the condition provides an array of answers and the current answer is among then
-						(Array.isArray(props.questionnaireItemMap[condition.question].answer) && props.questionnaireItemMap[condition.question].answer.includes( condition[getEnableWhenAnswerType(condition)]))
-						||
-						// OR: there is only one answer and it matches
-						(getCorrectlyFormattedAnswer(props.questionnaireItemMap[condition.question]) === condition[getEnableWhenAnswerType(condition)])
-					) {
-						// in case a matching condition is found
-						returnValue = true
+		} else {
+			if (item.enableWhen.length !== 0) {
+				let codingEquals = (coding1, coding2) => {
+					return coding1.display === coding2.display; //assuming display is always unique TODO: Fallback to code comparison in other cases
+				}
```

Comment:
I just noticed: Since this is ultimately the same as [this method](https://github.com/NUMde/compass-numapp-frontend/blob/430244a91d291dc4a509e988f4f570fff2dbea43/src/components/modal/questionnaireModal.js#L328) i think we should place the method in a shared spot @GolexPrime

Reply:
absolutely - `questionnaireAnalyzer.js ` itself might be a good place

Reply:
I outsourced this method in the latest commit